crypto/types/types.proto

```diff
@@ -3,49 +3,36 @@ package cosmos_sdk.crypto.v1;
 
 import "third_party/proto/gogoproto/gogo.proto";
 import "google/protobuf/any.proto";
+import "types/types.proto";
 
 option go_package = "github.com/cosmos/cosmos-sdk/crypto/types";
 
 // PublicKey specifies a public key
 message PublicKey {
-    // sum specifies which type of public key is wrapped
-    oneof sum {
-        bytes secp256k1 = 1;
-        bytes ed25519 = 2;
-        bytes sr25519 = 3;
-        PubKeyMultisigThreshold multisig = 4;
-        bytes secp256r1 = 5;
-
-        // any_pubkey can be used for any pubkey that an app may use which is
-        // not explicitly defined in the oneof
-        google.protobuf.Any any_pubkey = 15; // 15 is largest field that occupies one byte
-    }
+  // sum specifies which type of public key is wrapped
+  oneof sum {
+    bytes                   secp256k1 = 1;
+    bytes                   ed25519   = 2;
+    bytes                   sr25519   = 3;
+    PubKeyMultisigThreshold multisig  = 4;
+    bytes                   secp256r1 = 5;
+
+    // any_pubkey can be used for any pubkey that an app may use which is
+    // not explicitly defined in the oneof
+    google.protobuf.Any any_pubkey = 15;  // 15 is largest field that occupies one byte
+  }
 }
 
 // PubKeyMultisigThreshold specifies a public key type which nests multiple public
 // keys and a threshold
 message PubKeyMultisigThreshold {
-    uint32 threshold = 1 [(gogoproto.customname) = "K", (gogoproto.moretags) = "yaml:\"threshold\""];
-    repeated PublicKey pubkeys = 2 [(gogoproto.customname) = "PubKeys", (gogoproto.moretags) = "yaml:\"pubkeys\""];
+  uint32             threshold = 1 [(gogoproto.customname) = "K", (gogoproto.moretags) = "yaml:\"threshold\""];
+  repeated PublicKey pubkeys   = 2 [(gogoproto.customname) = "PubKeys", (gogoproto.moretags) = "yaml:\"pubkeys\""];
 }
 
-
 // MultiSignature wraps the signatures from a PubKeyMultisigThreshold.
 // See cosmos_sdk.tx.v1.ModeInfo.Multi for how to specify which signers signed
 // and with which modes
 message MultiSignature {
```

Comment:
So we now have two `MultiSignature` proto types. This can certainly lead to confusion. What are the differences between the two and when use one over the other? The godocs should reflect this.

Reply:
I brought back the hand-written type as I am not sure how this should be integrated to provide backwards compatability.

Reply:
@aaronc? Sorry, I'm a bit out of touch with the proto work with how it relates to keys...

Reply:
@alexanderbez are you referring to the `ModeInfo.Multi` and `Multisignature`? One is for the mode infos and the other is the actual signatures.... Is that what you mean?

Reply:
No, just `Multisignature`.

Reply:
I'm confused. There are two types?

Reply:
Yes -- review the PR here. There are `types.Multisignature` and `multisig.Multisignature`

Reply:
I think it's fine for now. I'm going to need to rework some of this stuff to support multiple signing modes in #6216. In the meantime, maybe we can rename `multisig.Multisignature` to something different?